api/helpers/game/emit-game-state.js

```diff
@@ -0,0 +1,60 @@
+import GameMoveType from '../../../utils/GameMoveType.json';
+
+module.exports = {
+  friendlyName: 'Emit Game State',
+
+  description: 'Determines if the game has ended and sends a socket event based on the game status.',
+
+  inputs: {
+    game: {
+      type: 'ref',
+      description: 'Game overview object',
+      required: true,
+    },
+    gameState: {
+      type: 'ref',
+      description: 'Game state object',
+      required: true,
+    },
+  },
+
+  fn: async function ({ game, gameState }, exits) {
+    let victory;
+
+    const lastEventChange = Object.keys(GameMoveType).find((key) => GameMoveType[key] === gameState.moveType);
+    const lastEventOneOffRank = lastEventChange === 'resolve' ? gameState.oneOff?.rank : null;
+    const lastEventCardChosen = lastEventChange === 'resolveThree' ? gameState.chosenCard : null;
+    const lastEventPlayerChoosing = lastEventChange === 'resolveThree' ? gameState.playedBy : null;
+    const lastEventDiscardedCards = lastEventChange === 'resolveFour' ? gameState.discardedCards : null;
+
+    const populatedGame = {
+      ...game,
+      ...gameState,
+      players: [
+        { ...game.p0, ...gameState.p0 },
```

Comment:
We need to be careful not to over-include from the User record itself, as we don't want to include the users' `encryptedPassword`s. Check the `formatPlayerData` in the `gameService`